src/libraries/System.Private.CoreLib/src/System/Globalization/CompareInfo.cs

```diff
@@ -915,11 +907,6 @@ public virtual unsafe int IndexOf(string source, char value, int startIndex, int
                 throw new ArgumentOutOfRangeException(nameof(count), SR.ArgumentOutOfRange_Count);
             }
 
-            if (source.Length == 0)
-            {
```

Comment:
I believe in such cases we should just compare value string against "" and return right away. no need to go further. This should be done after validating the flags in the next if block.